astropy/io/fits/column.py

```diff
@@ -1227,7 +1227,7 @@ def _convert_to_valid_data_type(self, array):
                         fsize = dims[-1]
                     else:
                         fsize = np.dtype(format.recformat).itemsize
-                    return chararray.array(array, itemsize=fsize)
+                    return chararray.array(array, itemsize=fsize, copy=False)
```

Comment:
So this is the only real change in `io.fits` when *creating* the BinTableHDU object (not taking into account the bytes decoding which happens only when accessing a bytes column). I'm trying to do the same but in my case it seems that the data is still loaded in memory whereas your memory usage report shows that you fully use the memmap. Any idea why, or am I missing something ?

Reply:
Did you explicitly set ``return_bytes=True`` when opening the file? Or you mean creating a new HDU object from scratch? copy=False isn't guaranteed to never copy, it might copy if the item size changes for example - can you share your specific example?

Reply:
I was trying just reading a file directly with `io.fits` and looking at the memory usage, independently from `Table`. And I found the culprit, it's because of a boolean column: just below the chararray block, for the 'L' format the array is read to check if False or 0 values must be converted.

Reply:
Perhaps worthy of raising as a separate issue; no reason this couldn't be done when truly needed.